modules/programs/tmux.nix

```diff
@@ -23,14 +51,125 @@ let
     };
   };
 
+  statusModule = types.submodule {
+    options = {
+      currentWindowFormat = mkOption {
+        default = "";
+        type = types.str;
+        description = ''
+          Current window status formatting.
+        '';
+      };
+
+      currentWindowStyle = mkOption {
+        default = "";
+        type = types.str;
+        description = ''
+          Current window style.
+        '';
+      };
+
+      justification = mkOption {
+        default = "centre";
+        type = types.str;
+        description = ''
+          How to justify statusbar info.
+        '';
+      };
+
+      keyMode = mkOption {
+        default = defaultKeyMode;
+        type = types.str;
+        description = ''
+          VI or Emacs style shortcuts.
+        '';
+      };
+
+      leftFormat = mkOption {
+        default = "";
+        type = types.str;
+        description = ''
+          Left status part formatting.
+        '';
+      };
+
+      leftLength = mkOption {
+        default = 30;
+        type = types.int;
+        description = ''
+          Length of left status part.
+        '';
+      };
+
+      messageStyle = mkOption {
+        default = "";
+        type = types.str;
+        description = ''
+          Message area style.
+        '';
+      };
+
+      rightFormat = mkOption {
+        default = "";
+        type = types.str;
+        description = ''
+          Left status part formatting.
+        '';
+      };
+
+      rightLength = mkOption {
+        default = 140;
+        type = types.int;
+        description = ''
+          Length of right status part.
+        '';
+      };
+
+      style = mkOption {
+        default = "";
+        type = types.str;
+        description = ''
+          Style.
+        '';
+      };
+
+      updateInterval = mkOption {
+        default = 1;
+        type = types.int;
+        description = ''
+          Seconds to pass between statusbar updates.
+        '';
+      };
+
+      windowFormat = mkOption {
+        default = "";
+        type = types.str;
+        description = ''
+          Window status formatting.
+        '';
+      };
+
+      windowStyle = mkOption {
+        default = "";
+        type = types.str;
+        description = ''
+          Current style.
+        '';
+      };
+    };
+  };
+
   defaultKeyMode  = "emacs";
   defaultResize   = 5;
-  defaultShortcut = "b";
+  defaultShell    = "$SHELL";
+  defaultShortcut = if versionAtLeast config.home.stateVersion "19.09"
+                    then "C-b" else "b";
   defaultTerminal = "screen";
 
   boolToStr = value: if value then "on" else "off";
 
   tmuxConf = ''
+    set  -g default-shell "${cfg.shell}"
```

Comment:
I tried installing tmux and reading the documentation. From what I can tell this option is intended for use when tmux is used as a login shell and it should be set to the absolute path of the executable. And by default it will fall back to /bin/sh but I guess only if you don't explicitly set the `default-shell` option?

From what I can tell this is a niche option. I would suggest not attempting to manage this option in Home Manager.